x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/dashboards/api/util/installation.ts

```diff
@@ -0,0 +1,186 @@
+    installationErrors.push(...errors);
+    await dashboardMigrationsClient.data.items.update(dashboardsToUpdate);
+    dashboardsToInstall = await dashboardBatches.next();
+  }
+
+  // Throw an error if needed
+  if (installationErrors.length) {
+    throw new Error(installationErrors.map((err) => err.message).join(', '));
+  }
+
+  return installedCount;
+};
+
+const installDashboards = async (
+  dashboardsToInstall: DashboardMigrationDashboard[],
+  savedObjectsClient: SavedObjectsClientContract,
+  spaceId: string
+): Promise<{
+  dashboardsToUpdate: Array<DashboardMigrationDashboard>;
+  errors: Error[];
+}> => {
+  const errors: Error[] = [];
+  const dashboardsToUpdate: Array<DashboardMigrationDashboard> = [];
+
+  const createDashboardsOutcome = await initPromisePool({
+    concurrency: MAX_DASHBOARDS_TO_CREATE_IN_PARALLEL,
+    items: dashboardsToInstall,
+    executor: async (dashboard) => {
+      try {
+        // Check if dashboard data exists
+        if (!dashboard.elastic_dashboard?.data) {
+          throw new Error('Invalid Dashboard: No data to install');
+        }
+
+        // Parse the dashboard data (assuming it's JSON)
+        const dashboardData = JSON.parse(dashboard.elastic_dashboard.data);
+        const tagNames = [getVendorTag(dashboard.original_dashboard.vendor)];
+
+        // Find or create tag IDs by name
+        const tagReferences = [];
+        for (const tagName of tagNames) {
+          let tagResults = await findTagsByName({
+            savedObjectsClient,
+            tagName,
+          });
+
+          // If tag doesn't exist, create it
+          if (tagResults.length === 0) {
+            try {
+              const createdTag = await createTag({
+                savedObjectsClient,
+                tagName,
+                description: `Auto-created tag for ${tagName}`,
+              });
+              // Convert SavedObject to SavedObjectsFindResult format
+              tagResults = [
+                {
+                  ...createdTag,
+                  score: 0,
+                },
+              ];
+            } catch (createError) {
+              // For now, we'll skip the tag silently
+            }
+          }
+
+          if (tagResults.length > 0) {
+            tagReferences.push({
+              id: tagResults[0].id,
+              name: `tag-ref-${tagName}`,
+              type: 'tag',
+            });
+          }
+        }
+
+        const result = await savedObjectsClient.create(
+          'dashboard',
+          {
+            title: dashboard.original_dashboard.title,
+            description: dashboard.original_dashboard.description,
+            // Add other dashboard attributes as needed
+            ...dashboardData.attributes,
+          },
+          {
+            id: dashboard.id,
+            references: tagReferences,
+            initialNamespaces: [spaceId],
+          }
+        );
+
+        dashboardsToUpdate.push({
+          ...dashboard,
```

Comment:
It's probably better to set only the `id` here, just like we do for rules
```suggestion
          id: dashboard.id,
```